app/payment-link-v2/reference-confirm/reference-confirm.njk

```diff
@@ -35,14 +36,14 @@
     <div class="govuk-button-group">
       {{ govukButton({
         text: __p('paymentLinksV2.referenceConfirm.confirmAndContinue'),
-        href: confirmPageUrl,
+        href: amountPageUrl,
```

Comment:
The next page will be amount, only if the product doesn't have a fixed price. If it does have a fixed price, the next page should be confirm. The logic for this should be the same as for the reference page.